src/main/java/xbot/common/subsystems/drive/SwerveBezierTrajectoryCommand.java

```diff
@@ -0,0 +1,98 @@
+package xbot.common.subsystems.drive;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import xbot.common.logging.RobotAssertionManager;
+import xbot.common.properties.PropertyFactory;
+import xbot.common.subsystems.drive.control_logic.HeadingModule;
+import xbot.common.subsystems.pose.BasePoseSubsystem;
+import xbot.common.trajectory.XbotSwervePoint;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A "not-so-simple" SwerveSimpleTrajectoryCommand that does a Bézier curve!~~
+ * TO USE: call setBezierConfiguration and pass it the needed params
+ * If you are EXTENDING this because you need some logic to compute control points:
+ * Remember to call setBezierConfiguration and super.initialize()
+ */
+public class SwerveBezierTrajectoryCommand extends SwerveSimpleTrajectoryCommand {
+
+    List<Translation2d> controlPoints;
+    Pose2d endPoint;
+    int steps;
+
+    @Inject
+    public SwerveBezierTrajectoryCommand(BaseSwerveDriveSubsystem drive, BasePoseSubsystem pose, PropertyFactory pf,
+                                         HeadingModule.HeadingModuleFactory headingModuleFactory, RobotAssertionManager assertionManager) {
+        super(drive, pose, pf, headingModuleFactory, assertionManager);
+    }
+
+    @Override
+    public void initialize() {
+        setBezierCurve(controlPoints, endPoint, steps);
+        super.initialize();
+    }
+
+    /**
+     * Set the configuration of the Bézier curve, we can't generate our curve immediately as our pose will change
+     * @param controlPoints of the Bézier curve
+     * @param endPoint of our command
+     * @param steps to split our Bézier curve into
+     */
+    public void setBezierConfiguration(List<Translation2d> controlPoints, Pose2d endPoint, int steps) {
+        this.controlPoints = controlPoints;
+        this.steps = steps;
+        this.endPoint = endPoint;
+    }
+
+    /**
+     * Generates a Bézier curve for our command!
+     */
+    private void setBezierCurve(List<Translation2d> controlPoints, Pose2d endPoint, int steps) {
+        List<XbotSwervePoint> bezierPoints = new ArrayList<>();
+        List<Translation2d> allPoints = new ArrayList<>();
+        allPoints.add(pose.getCurrentPose2d().getTranslation());
+        allPoints.addAll(controlPoints);
+        allPoints.add(endPoint.getTranslation());
+
+        Rotation2d startingRotation = pose.getCurrentPose2d().getRotation();
+        for (int i = 1; i <= steps; i++) {
+            double lerpFraction = i / (double) steps;
+            XbotSwervePoint point = new XbotSwervePoint(
+                    deCasteljauIterative(allPoints, lerpFraction),
```

Comment:
Highly recommend watching 
- https://youtu.be/aVwxzDHniEw?si=ilXv-GDybxG6sze2
- https://youtu.be/jvPPXbo87ds?si=gbLHxpY7ISI2sOsC
-

Reply:
Both videos are worth watching in their entirety, but the first one has a section about deCasteljau early on